hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -379,16 +379,29 @@ public boolean hasTargetArtifactAssigned(final long targetId, final String sha1H
     @Transactional(isolation = Isolation.READ_COMMITTED)
     @Retryable(include = ConcurrencyFailureException.class, exclude = EntityAlreadyExistsException.class, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
     public Target findOrRegisterTargetIfItDoesNotExist(final String controllerId, final URI address) {
+        String name = controllerId;
+        return findOrRegisterTargetIfItDoesNotExist(controllerId, address, name);
+    }
+
+    @Override
+    @Transactional(isolation = Isolation.READ_COMMITTED)
+    @Retryable(include = ConcurrencyFailureException.class, exclude = EntityAlreadyExistsException.class, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
+    public Target findOrRegisterTargetIfItDoesNotExist(final String controllerId, final URI address,
+            final String name) {
         final Specification<JpaTarget> spec = (targetRoot, query, cb) -> cb
                 .equal(targetRoot.get(JpaTarget_.controllerId), controllerId);
 
-        return targetRepository.findOne(spec).map(target -> updateTargetStatus(target, address))
-                .orElseGet(() -> createTarget(controllerId, address));
+        return targetRepository.findOne(spec).map(target -> updateTarget(target, address, name))
+                .orElseGet(() -> createTarget(controllerId, address, name));
     }
 
-    private Target createTarget(final String controllerId, final URI address) {
+    private Target createTarget(final String controllerId, final URI address, String name) {
+        // In case of a true expression, the targetId will be set as name
+        if (!StringUtils.hasText(name)) {
+            name = controllerId;
```

Comment:
This check/assignment was introduced after the first peer review, as it was not nice to have two separated createTarget() methods (one for case with provided name property, one with defaultName=controllerId without name parameter). Would you mind making a suggestion how to avoid using the check & assignment, but still only use one createTarget() method?

Reply:
Sorry, didn't see you suggestion below - thanks for the hint! 👍

Reply:
Done.